apps/shared/avifutil.h

```diff
@@ -10,6 +10,59 @@
 extern "C" {
 #endif
 
+// NEXTARG(), MAIN(), INIT_ARGV(), FREE_ARGV() for UTF8-aware command line parsing.
+#if defined(_WIN32)
+#define NEXTARG()                                                      \
+    if (((argIndex + 1) == argc) || (argv[argIndex + 1][0] == L'-')) { \
+        fprintf(stderr, "%s requires an argument.", arg);              \
+        goto cleanup;                                                  \
+    }                                                                  \
+    arg = argv[++argIndex]
+#else
+#define NEXTARG()                                                     \
+    if (((argIndex + 1) == argc) || (argv[argIndex + 1][0] == '-')) { \
+        fprintf(stderr, "%s requires an argument.", arg);             \
+        goto cleanup;                                                 \
+    }                                                                 \
+    arg = argv[++argIndex]
+#endif
+
+#if defined(_WIN32)
+#define MAIN() int wmain(int argc, wchar_t * _argv[])
```

Comment:
Nit: I suggest using `wargv` instead of `_argv`. I missed the initial underscore when I first read it.

Note: symbols that start with `_` or `__` are reserved for the C implementation. So that's another reason to avoid `_argv`.

Reply:
Done